c10/core/TensorImpl.h

```diff
@@ -2638,6 +2651,14 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   // direction (to make sure the pyobj stays live).
   bool owns_pyobj_ : 1;
 
+  // Typically, the tensor dispatch keys define the tensor layout
+  // uniquely. This allows using non-virtual layout method for better
+  // performance. However, when tensor's layout depends, say, on
+  // tensor attributes, set define_layout_from_dispatch_key to
+  // false. In this case, the corresponding tensor impl class must
+  // overwrite virtual layout_impl() method.
+  bool define_layout_from_dispatch_key_ : 1;
```

Comment:
We don't need this extra bit though, at least for now. It's more general, but it'll impose additional memory cost.

Reply:
Removed.